Purchases/Public/RCPurchases.m

```diff
@@ -166,7 +168,7 @@ - (instancetype)initWithAppUserID:(NSString *)appUserID
         self.productsByIdentifier = [NSMutableDictionary new];
         self.purchaseCompleteCallbacks = [NSMutableDictionary new];
 
-        self.finishTransactions = YES;
+        self.finishTransactions = !observerMode;
```

Comment:
Should finishTransactions be set before the storeKitWrapper in case there are already transactions on the queue?

Reply:
It's set before we set the storekitwrapper delegate so we should be fine